webviz_subsurface_components/py_expression_eval.py

```diff
@@ -0,0 +1,483 @@
+                    self.addfunc(tokenstack, operstack, TFUNCALL)
+                expected = (
+                    self.PRIMARY
+                    | self.LPAREN
+                    | self.FUNCTION
+                    | self.SIGN
+                    | self.NULLARY_CALL
+                )
+            elif self.isRightParenth():
+                if expected & self.NULLARY_CALL:
+                    token = Token(TNUMBER, 0, 0, [])
+                    tokenstack.append(token)
+                elif (expected & self.RPAREN) == 0:
+                    self.error_parsing(self.pos, 'unexpected ")"')
+                expected = self.OPERATOR | self.RPAREN | self.LPAREN | self.CALL
+            elif self.isConst():
+                if (expected & self.PRIMARY) == 0:
+                    self.error_parsing(self.pos, "unexpected constant")
+                consttoken = Token(TNUMBER, 0, 0, self.tokennumber)
+                tokenstack.append(consttoken)
+                expected = self.OPERATOR | self.RPAREN
+            elif self.isOp2():
+                if (expected & self.FUNCTION) == 0:
+                    self.error_parsing(self.pos, "unexpected function")
+                self.addfunc(tokenstack, operstack, TOP2)
+                noperators += 2
+                expected = self.LPAREN
+            elif self.isOp1():
+                if (expected & self.FUNCTION) == 0:
+                    self.error_parsing(self.pos, "unexpected function")
+                self.addfunc(tokenstack, operstack, TOP1)
+                noperators += 1
+                expected = self.LPAREN
+            elif self.isVar():
+                if (expected & self.PRIMARY) == 0:
+                    self.error_parsing(self.pos, "unexpected variable")
+                vartoken = Token(TVAR, self.tokenindex, 0, 0)
+                tokenstack.append(vartoken)
+                expected = self.OPERATOR | self.RPAREN | self.CALL
+            elif self.isWhite():
+                pass
+            else:
+                if self.errormsg == "":
+                    self.error_parsing(self.pos, "unknown character")
+                else:
+                    self.error_parsing(self.pos, self.errormsg)
+        if self.tmpprio < 0 or self.tmpprio >= 10:
+            self.error_parsing(self.pos, 'unmatched "()"')
+        while len(operstack) > 0:
+            tmp = operstack.pop()
+            tokenstack.append(tmp)
+        if (noperators + 1) != len(tokenstack):
+            self.error_parsing(self.pos, "parity")
+
+        return Expression(tokenstack, self.ops1, self.ops2)
+
+    def evaluate(self, expr, variables):
+        return self.parse(expr).evaluate(variables)
+
+    def error_parsing(self, column, msg):
+        self.success = False
+        self.errormsg = (
+            "parse error [column "
+            + str(column)
+            + "]: "
+            + msg
+            + ", expression: "
+            + self.expression
+        )
```

Comment:
Strings like this are nice and quick to convert to f-strings since we only support Python 3+.
```suggestion
        self.errormsg = f"parse error [column {column}]: {msg}, expression: {self.expression}"
```